webpack/translations-plugin.js

```diff
@@ -0,0 +1,128 @@
+const path = require('path')
+const jsyaml = require('js-yaml')
+const fs = require('fs')
+const JS_INDENT = 2
+const FILENAME = 'en'
+
+const marketplaceKeys = [
+  'name',
+  'description',
+  'short_description',
+  'long_description',
+  'installation_instructions',
+  'parameters'
+]
+
+function toObj (yaml) {
+  const obj = jsyaml.safeLoad(yaml)
+  const pkg = obj.parts[0].translation.key.match(/^txt.apps.([^.]+)/)[1]
+
+  let translations = obj.parts.map(function (part) { return part.translation })
+  translations = translations.filter(function (translation) {
+    return translation.obsolete ? (new Date(translation.obsolete) > new Date()) : true
+  })
+
+  return arrayToNestedHash(translations).txt.apps[pkg]
+}
+
+function arrayToNestedHash (arr) {
+  return arr.reduce(function (result, item) {
+    const keys = item.key.split('.')
```

Comment:
The structure of the input array and output hash is not quite clear here. I think having some comments with expected input and output would be better to explain this function. Also, can you add some empty lines in between?

Reply:
👍 Done.